app/controllers/images_controller.rb

```diff
@@ -1,12 +1,16 @@
 class ImagesController < ApplicationController
   before_action :require_user
   include Glitchable
+  include S3able
+
   def create
-    image = current_user.api_images.create!(image_params)
-    download = Down.download(image.url)
-    image.original.attach(io: download, filename: "#{image.url}.jpg")
-    basic_glitch(image)
-    redirect_to search_show_path(image)
+    api_image = current_user.api_images.create!(image_params)
+    download = Down.download(api_image.url)
+    api_image.original.attach(io: download, filename: "#{api_image.url}.jpg")
+    basic_glitch(api_image)
+    api_image.attach_key(upload_image)
```

Comment:
The return value of upload_image is a unique key which can then be called to get the location of the image stored by S3.